internal/controller/nnf_workflow_controller_helpers.go

```diff
@@ -713,37 +713,86 @@ func (r *NnfWorkflowReconciler) getLustreMgsFromPool(ctx context.Context, pool s
 		return corev1.ObjectReference{}, "", dwsv1alpha2.NewResourceError("").WithUserMessage("no MGSs found for pool: %s", pool).WithFatal().WithUser()
 	}
 
-	persistentStorage := persistentStorageList.Items[rand.Intn(len(persistentStorageList.Items))]
+	healthyMgts := make(map[string]corev1.ObjectReference)
+	for _, persistentStorage := range persistentStorageList.Items {
+		// Find the NnfStorage for the PersistentStorage so we can check it's status and get the MGT LNid
+		nnfStorage := &nnfv1alpha1.NnfStorage{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      persistentStorage.Name,
+				Namespace: persistentStorage.Namespace,
+			},
+		}
 
-	// Find the NnfStorage for the PersistentStorage so we can get the LNid
-	nnfStorage := &nnfv1alpha1.NnfStorage{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      persistentStorage.Name,
-			Namespace: persistentStorage.Namespace,
-		},
-	}
+		if err := r.Get(ctx, client.ObjectKeyFromObject(nnfStorage), nnfStorage); err != nil {
+			return corev1.ObjectReference{}, "", dwsv1alpha2.NewResourceError("could not get persistent NnfStorage %v for MGS", client.ObjectKeyFromObject(nnfStorage)).WithError(err)
+		}
 
-	if err := r.Get(ctx, client.ObjectKeyFromObject(nnfStorage), nnfStorage); err != nil {
-		return corev1.ObjectReference{}, "", dwsv1alpha2.NewResourceError("could not get persistent NnfStorage %v for MGS", client.ObjectKeyFromObject(nnfStorage)).WithError(err)
-	}
+		// Do some sanity checks on the NnfStorage to make sure it's really a standalone MGT
+		if nnfStorage.Spec.FileSystemType != "lustre" {
```

Comment:
Do we have some other place where these sanity checks happen?
I think you should log it, and then go to the next one rather than returning Fatal.
Maybe check nnfStorage.Status.Ready first, rather than last.

Reply:
If these checks fail, then something is seriously wrong. These are the only values that the nnfstorage can have if it was set up as a Lustre MGT. If the values don't match, then someone must have created an nnfstorage manually.